vnext/Microsoft.ReactNative/ReactHost/ReactContext.cpp

```diff
@@ -4,10 +4,99 @@
 #include "ReactContext.h"
 #include <winrt/Microsoft.ReactNative.h>
 #include "Microsoft.ReactNative/IReactNotificationService.h"
+#include "MsoUtils.h"
 #include "ReactInstanceWin.h"
 
 namespace Mso::React {
 
+//=============================================================================================
+// ReactSettingsSnapshot implementation
+//=============================================================================================
+
+#ifndef CORE_ABI // requires instance
+
+ReactSettingsSnapshot::ReactSettingsSnapshot(Mso::WeakPtr<ReactInstanceWin> &&reactInstance) noexcept
+    : m_reactInstance{std::move(reactInstance)} {}
+
+bool ReactSettingsSnapshot::UseWebDebugger() const noexcept {
+  if (auto instance = m_reactInstance.GetStrongPtr()) {
+    return instance->UseWebDebugger();
+  }
+  return false;
+}
+
+bool ReactSettingsSnapshot::UseFastRefresh() const noexcept {
+  if (auto instance = m_reactInstance.GetStrongPtr()) {
+    return instance->UseFastRefresh();
+  }
+  return false;
+}
+
+bool ReactSettingsSnapshot::UseDirectDebugger() const noexcept {
+  if (auto instance = m_reactInstance.GetStrongPtr()) {
+    return instance->UseDirectDebugger();
+  }
+  return false;
+}
+
+bool ReactSettingsSnapshot::DebuggerBreakOnNextLine() const noexcept {
+  if (auto instance = m_reactInstance.GetStrongPtr()) {
+    return instance->DebuggerBreakOnNextLine();
+  }
+  return false;
+}
+
+uint16_t ReactSettingsSnapshot::DebuggerPort() const noexcept {
+  if (auto instance = m_reactInstance.GetStrongPtr()) {
+    return instance->DebuggerPort();
+  }
+  return 0;
+}
+
+std::string ReactSettingsSnapshot::DebugBundlePath() const noexcept {
+  if (auto instance = m_reactInstance.GetStrongPtr()) {
+    return instance->DebugBundlePath();
+  }
+  return {};
+}
+
+std::string ReactSettingsSnapshot::BundleRootPath() const noexcept {
+  if (auto instance = m_reactInstance.GetStrongPtr()) {
+    return instance->BundleRootPath();
+  }
+  return {};
+}
+
+std::string ReactSettingsSnapshot::SourceBundleHost() const noexcept {
+  if (auto instance = m_reactInstance.GetStrongPtr()) {
+    return instance->SourceBundleHost();
+  }
+  return {};
+}
+
+uint16_t ReactSettingsSnapshot::SourceBundlePort() const noexcept {
+  if (auto instance = m_reactInstance.GetStrongPtr()) {
+    return instance->SourceBundlePort();
+  }
+  return 0;
```

Comment:
I know this is what the code did before, but it seems like this wouldn't ever do what we wanted if the instance died. Might be worth crashing instead if we can't give the right answer and don't want folks querying settings after instance destruction. 

If we have the snapshot as its own immutable object which is small, another option could be to just cache settings as members on the snapshot.

Reply:
I want to have the API changes before we cut the 0.64 stable branch. I will address your point in the follow up PR.